pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/BufferedData.java

```diff
@@ -645,31 +671,38 @@ public double readDouble(@NonNull final ByteOrder byteOrder) {
      */
     @Override
     public int readVarInt(final boolean zigZag) {
+        if (direct) return ReadableSequentialData.super.readVarInt(zigZag);
```

Comment:
I assume this is making us slower for direct buffers, till we add a subclass for direct buffers that uses unsafe or varhandles.

Reply:
Yes!

Reply:
That is the intent.